inc/user.class.php

```diff
@@ -4854,7 +4882,11 @@ public function updateForgottenPassword(array $input) {
          }
 
       } else {
-         throw new ForgetPasswordException(__('Email address not found.'));
+         if (self::countUsersByEmail($input['email'], $condition) > 1) {
+            throw new ForgetPasswordException(__('Multiple users were found for this email address. Please contact your administrator.'));
+         } else {
+            throw new ForgetPasswordException(__('Email address not found.'));
+         }
```

Comment:
It is generally a bad idea to expose the fact that an account exists for a given email, as it can help to identify login / emails of the user base.
May be the right way to do would be to send the recovery email to all users having the email. The same user would receive then multiple emails, and maybe one of them would contain informations that can help him to identify which recovery link to use.

Reply:
I'm not sure sending several emails is the solution; it's not sure the related account really belongs to the attached email; and this should also cause issues to enable the wrong account on some cases I guess.

Anyways, I'm OK to stay "vague" in our error messages related to email and authentication. We may add an info in logs; so admin can know what went wrong.